File: array/1_TwoSum/1_TwoSum.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

   	result := []int {-1,-1}

   	for i := 0; i < len(nums); i++ {

	   item := nums[i]
	   var diff int = target - item
   
	   if m[diff] > 0 {
		   result[0] = m[diff] - 1
		   result[1]=  i
		   break;
	   }

	   m[item] = i+1;
   	}
   
   	return result
}

File: array/1_TwoSum/1_TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_dict = {}
        result = []
        for i in range(0, len(nums)):
            
            item = nums[i]
            diff = target - nums[i]
           
            if my_dict.get(diff, -1) > -1:
                result.append(i)
                result.append(my_dict[diff])

            my_dict[item] = i 
        
        return result